src/backend/distributed/utils/maintenanced.c

```diff
@@ -200,11 +203,17 @@ InitializeMaintenanceDaemonBackend(void)
 
 		if (!RegisterDynamicBackgroundWorker(&worker, &handle))
 		{
-			ereport(ERROR, (errmsg("could not start maintenance background worker"),
-							errhint("Increasing max_worker_processes might help.")));
+			ereport(WARNING, (errmsg("could not start maintenance background worker"),
```

Comment:
Here the reason registration failed is similar to the case here right ? : https://github.com/citusdata/citus/pull/3815/files#diff-165406aafb69addc03fd1397f9545a6aR156

If so, it might make sense to put this to a method and call it such as `WarnMaintenanceNotStarted`. (as we are giving the same warning, instead of having it in two different places).

Reply:
done